src/PlatesRendererFactory.php

```diff
@@ -70,13 +49,21 @@ public function __invoke(ContainerInterface $container): PlatesRenderer
      * Otherwise, invokes the PlatesEngineFactory with the $container to create
      * and return the instance.
      */
-    private function createEngine(ContainerInterface $container): PlatesEngine
+    private function getEngine(ContainerInterface $container): PlatesEngine
     {
         if ($container->has(PlatesEngine::class)) {
             return $container->get(PlatesEngine::class);
         }
 
-        $engineFactory = new PlatesEngineFactory();
-        return $engineFactory($container);
+        trigger_error(sprintf(
```

Comment:
IMO, this execution branch can be dropped completely.

We already provide service configuration that works with the declared service.

Reply:
Then I think we can do the same here:

 - https://github.com/mezzio/mezzio-twigrenderer/blob/cfe9ffb3521dab22cafa8e4c3b5865af0f117918/src/TwigRendererFactory.php#L89-L98